javers-core/src/test/groovy/org/javers/core/metamodel/type/MapTypeTest.groovy

```diff
@@ -63,4 +66,17 @@ class MapTypeTest extends Specification{
         mType.keyType == String
         mType.valueType == Integer
     }
+
+    def "should return key & value Class if baseJavaType and value type are generic"(){
```

Comment:
Ok, MapTypeTest is successful but TypeMapperTest fails without the fix in ParametrizedDehydratedType. Anyway, I've modified SnapshotRepositry and JaversRepositoryE2ETest to highlight the actual problem.